src/builder-manifest.c

```diff
@@ -2844,18 +2887,23 @@ builder_manifest_finish (BuilderManifest *self,
           g_autofree char *extension_metadata_name = NULL;
           g_autoptr(GFile) metadata_extension_file = NULL;
           g_autofree char *metadata_contents = NULL;
+          g_autofree char *extension_tag_opt = NULL;
 
           if (!builder_extension_is_bundled (e))
             continue;
 
+          extension_tag_opt = maybe_format_extension_tag (builder_manifest_get_extension_tag (self));
           extension_metadata_name = g_strdup_printf ("metadata.%s", builder_extension_get_name (e));
           metadata_extension_file = g_file_get_child (app_dir, extension_metadata_name);
           metadata_contents = g_strdup_printf ("[Runtime]\n"
                                                "name=%s\n"
                                                "\n"
                                                "[ExtensionOf]\n"
-                                               "ref=%s\n",
-                                               builder_extension_get_name (e), ref);
+                                               "ref=%s\n"
+                                               "%s\n",
```

Comment:
Can you add the \n in maybe_format_extension_tag so we don't leave an unnecessary \n here otherwise.